var/spack/repos/builtin/packages/icedtea/package.py

```diff
@@ -0,0 +1,163 @@
+    resource(name='openjdk', placement='openjdk_src',
+             sha512='f3cca223bd39c0202dd1a65a38ca17024b6cb5c82d833946ec1b7d28d205833b4dd2dadde505a1c2384e3b28ff0d21a4f175e064b8ac82aa8a07508e53cdc722',
+             url='http://icedtea.wildebeest.org/download/drops/icedtea8/3.4.0/openjdk.tar.xz',
+             when='@3.4.0')
+    resource(name='nashorn', placement='nashorn_src',
+             sha512='79b5095bab447d1911696bc1e328fb72c08764c0139cab14a28c0f6c2e49a2d96bb06fbbb85523b2586672cb0f13709c3158823d5ac3f3fe3f0f88402d3cb246',
+             url='http://icedtea.wildebeest.org/download/drops/icedtea8/3.4.0/nashorn.tar.xz',
+             when='@3.4.0')
+    resource(name='shenandoah', placement='shenandoah_src',
+             sha512='0f085e87c63679314ef322b3f4b854792d46539d5530dd75a7fd45b8b6d663f58469be2808ea5fb4bf31f6c5369cb78f28e1599f748e1931ba7040136306eb20',
+             url='http://icedtea.wildebeest.org/download/drops/icedtea8/3.4.0/shenandoah.tar.xz',
+             when='@3.4.0')
+
+    def configure_args(self):
+        os.environ['POTENTIAL_CXX'] = os.environ['CXX']
+        os.environ['POTENTIAL_CC'] = os.environ['CC']
+        os.environ['WGET'] = self.spec['wget'].command.path
+        args = []
+        if '~X' in self.spec:
+            args.append('--enable-headless')
+        if '+shenandoah' in self.spec:
+            args.append('--with-hotspot-build=shenandoah')
+            args.append('--with-hotspot-src-zip=' + self.stage[9].archive_file)
+            args.append('--with-hotspot-checksum=no')
+        else:
+            args.append('--with-hotspot-src-zip=' + self.stage[2].archive_file)
+            args.append('--with-hotspot-checksum=no')
+        args += [
+            '--with-corba-src-zip=' + self.stage[1].archive_file,
+            '--with-cobra-checksum=no',
+            '--with-jaxp-src-zip=' + self.stage[3].archive_file,
+            '--with-jaxp-checksum=no',
+            '--with-jaxws-src-zip=' + self.stage[4].archive_file,
+            '--with-jaxws-checksum=no',
+            '--with-jdk-src-zip=' + self.stage[5].archive_file,
+            '--with-jdk-checksum=no',
+            '--with-langtools-src-zip=' + self.stage[6].archive_file,
+            '--with-langtools-checksum=no',
+            '--with-openjdk-src-zip=' + self.stage[7].archive_file,
+            '--with-openjdk-checksum=no',
+            '--with-nashorn-src-zip=' + self.stage[8].archive_file,
+            '--with-nashorn-checksum=no', '--disable-maintainer-mode'
+            '--disable-downloading', '--disable-system-pcsc',
+            '--disable-system-sctp', '--disable-system-kerberos',
+            '--with-jdk-home=' + self.spec['jdk'].prefix
+        ]
+        return args
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('JAVA_HOME', self.spec.prefix)
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        spack_env.set('JAVA_HOME', self.spec.prefix)
```

Comment:
For the record, `self.spec.prefix == self.prefix` if you want to make it shorter.